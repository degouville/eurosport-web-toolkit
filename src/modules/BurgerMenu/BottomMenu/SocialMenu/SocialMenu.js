import React from 'react';
import styled, { css } from 'react-emotion';
import PropTypes from 'prop-types';

import SocialIcon from '../../../../elements/SocialIcon';
import Link from '../../../../elements/Link';
import { white } from '../../../../colors';
import { large, medium } from '../../../../breakpoints';

const StyledWrapper = styled.div`
  color: ${white};
  display: flex;
  align-items: center;
  flex-grow: 0;
  flex-shrink: 0;
  height: auto;
  padding-top: 15px;

  ${medium(css`
    padding-top: 0;
  `)}
`;
const StyledTitle = styled.div`
  font-size: 12px;
  font-weight: bold;
`;
const StyledSocialItems = styled.ul`
  margin: 0;
  padding-left: 10px;
  list-style-type: none;
  display: flex;
  align-items: center;

  ${large(css`
    padding-left: 20px;
  `)}
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  color: ${white};
  vertical-align: middle;
  margin-right: 10px;
  font-size: 12px;

  ${large(css`
    margin-right: 20px;
  `)}
`;

const SocialMenu = ({ name, items, isMobile }) => {
  if (items.length === 0) return null;

  return (
    <StyledWrapper>
      <StyledTitle>{name}</StyledTitle>
      <StyledSocialItems>
        {items &&
          items.map(({ name: itemName, link, icon }) => (
            <li key={itemName}>
              <StyledLink href={link.url} target={link.blank ? '_blank' : '_self'}>
                <SocialIcon iconText={isMobile ? '' : itemName} iconType={icon} />
              </StyledLink>
            </li>
          ))}
      </StyledSocialItems>
    </StyledWrapper>
  );
};

SocialMenu.defaultProps = {
  isMobile: false,
};

SocialMenu.propTypes = {
  name: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      blank: PropTypes.bool,
      name: PropTypes.string,
      icon: PropTypes.string,
    })
  ).isRequired,
  isMobile: PropTypes.bool,
};

SocialMenu.displayName = 'SocialMenu';

export default SocialMenu;
