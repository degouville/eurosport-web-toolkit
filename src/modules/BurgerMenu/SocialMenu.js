import React from 'react';
import styled, { css } from 'react-emotion';
import PropTypes from 'prop-types';

import SocialIcon from '../../elements/SocialIcon';
import { large } from '../../breakpoints';
import { midnightExpress } from '../../colors';
import Link from '../../elements/Link';

const StyledWrapper = styled.div`
  background: ${midnightExpress};
  color: white;
  display: flex;
  align-items: center;
  height: 61px;
  padding-left: 22px;

  ${large(css`
    padding-left: 70px;
  `)};
`;
const StyledTitle = styled.div`
  font-size: 12px;
  font-weight: bold;
`;
const StyledSocialItems = styled.ul`
  margin: 0;
  padding-left: 10px;
  list-style-type: none;
  display: flex;
  align-items: center;
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  color: white;
  vertical-align: middle;
  margin-right: 10px;
  font-size: 12px;
`;

const SocialMenu = ({ name, items, isMobile }) => {
  if (items.length === 0) return null;

  return (
    <StyledWrapper>
      <StyledTitle>{name}</StyledTitle>
      <StyledSocialItems>
        {items &&
          items.map(({ name: itemName, link, icon }) => (
            <li key={itemName}>
              <StyledLink href={link.url} target={link.blank ? '_blank' : '_self'}>
                <SocialIcon iconText={isMobile ? '' : itemName} iconType={icon} />
              </StyledLink>
            </li>
          ))}
      </StyledSocialItems>
    </StyledWrapper>
  );
};

SocialMenu.defaultProps = {
  isMobile: false,
};

SocialMenu.propTypes = {
  name: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      blank: PropTypes.bool,
      name: PropTypes.string,
      icon: PropTypes.string,
    })
  ).isRequired,
  isMobile: PropTypes.bool,
};

export default SocialMenu;
