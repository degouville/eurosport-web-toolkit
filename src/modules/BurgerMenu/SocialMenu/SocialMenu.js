import React from 'react';
import styled, { css } from 'react-emotion';
import PropTypes from 'prop-types';

import { SocialIcon, Link } from '../../..';
import { large } from '../../../breakpoints';
import { coreLightMinus1, midnightExpress } from '../../../colors';

const StyledWrapper = styled.div`
  background: ${midnightExpress};
  color: ${coreLightMinus1};
  display: flex;
  align-items: center;
  height: 61px;
  padding-left: 22px;

  ${large(css`
    padding-left: 70px;
  `)};
`;
const StyledTitle = styled.div`
  font-size: 12px;
  font-weight: bold;
`;
const StyledSocialItems = styled.ul`
  margin: 0;
  padding-left: 10px;
  list-style-type: none;
  display: flex;
  align-items: center;
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  color: ${coreLightMinus1};
  vertical-align: middle;
  margin-right: 10px;
  font-size: 12px;
`;

const SocialMenu = ({ name, items, isMobile }) => {
  if (items.length === 0) return null;

  return (
    <StyledWrapper>
      <StyledTitle>{name}</StyledTitle>
      <StyledSocialItems>
        {items &&
          items.map(({ name: itemName, link, icon }) => (
            <li key={itemName}>
              <StyledLink href={link.url} target={link.blank ? '_blank' : '_self'}>
                <SocialIcon iconText={isMobile ? '' : itemName} iconType={icon} />
              </StyledLink>
            </li>
          ))}
      </StyledSocialItems>
    </StyledWrapper>
  );
};

SocialMenu.defaultProps = {
  isMobile: false,
};

SocialMenu.propTypes = {
  name: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      blank: PropTypes.bool,
      name: PropTypes.string,
      icon: PropTypes.string,
    })
  ).isRequired,
  isMobile: PropTypes.bool,
};

SocialMenu.displayName = 'SocialMenu';

export default SocialMenu;
