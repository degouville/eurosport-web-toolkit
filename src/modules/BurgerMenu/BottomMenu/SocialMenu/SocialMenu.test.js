import React from 'react';
import { shallow } from 'enzyme';
import SocialMenu from './SocialMenu';
import headerMock from '../../mocks/feed-menu';

const socialItems = headerMock.header.find(i => i.id === 7).sections[0].items;

it('renders with expected snapshot on desktop', () => {
  expect(shallow(<SocialMenu items={socialItems} name="follow us on socials" isMobile={false} />)).toMatchSnapshot();
});

it('renders with expected snapshot on mobile', () => {
  expect(shallow(<SocialMenu items={socialItems} name="follow us on socials" isMobile />)).toMatchSnapshot();
});

it('does not render when no items', () => {
  expect(shallow(<SocialMenu items={[]} name="follow us on socials" isMobile={false} />).type()).toEqual(null);
});
