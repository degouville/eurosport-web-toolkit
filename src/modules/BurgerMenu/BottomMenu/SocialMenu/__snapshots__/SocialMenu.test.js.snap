// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with expected snapshot on desktop 1`] = `
<Styled(div)>
  <Styled(div)>
    follow us on socials
  </Styled(div)>
  <Styled(ul)>
    <li
      key="Facebook"
    >
      <Styled(Link)
        href="//facebook.com/Eurosport-828777003809568"
        linkComponent={[Function]}
        target="_self"
      >
        <SocialIcon
          iconText="Facebook"
          iconType="facebook"
        />
      </Styled(Link)>
    </li>
    <li
      key="Twitter"
    >
      <Styled(Link)
        href="//twitter.com/Eurosport_UK"
        linkComponent={[Function]}
        target="_self"
      >
        <SocialIcon
          iconText="Twitter"
          iconType="twitter"
        />
      </Styled(Link)>
    </li>
    <li
      key="Snapchat"
    >
      <Styled(Link)
        href="//www.snapchat.com/add/eurosport"
        linkComponent={[Function]}
        target="_self"
      >
        <SocialIcon
          iconText="Snapchat"
          iconType="snapchat"
        />
      </Styled(Link)>
    </li>
    <li
      key="Instagram"
    >
      <Styled(Link)
        href="//www.instagram.com/eurosport"
        linkComponent={[Function]}
        target="_self"
      >
        <SocialIcon
          iconText="Instagram"
          iconType="instagram"
        />
      </Styled(Link)>
    </li>
  </Styled(ul)>
</Styled(div)>
`;

exports[`renders with expected snapshot on mobile 1`] = `
<Styled(div)>
  <Styled(div)>
    follow us on socials
  </Styled(div)>
  <Styled(ul)>
    <li
      key="Facebook"
    >
      <Styled(Link)
        href="//facebook.com/Eurosport-828777003809568"
        linkComponent={[Function]}
        target="_self"
      >
        <SocialIcon
          iconText=""
          iconType="facebook"
        />
      </Styled(Link)>
    </li>
    <li
      key="Twitter"
    >
      <Styled(Link)
        href="//twitter.com/Eurosport_UK"
        linkComponent={[Function]}
        target="_self"
      >
        <SocialIcon
          iconText=""
          iconType="twitter"
        />
      </Styled(Link)>
    </li>
    <li
      key="Snapchat"
    >
      <Styled(Link)
        href="//www.snapchat.com/add/eurosport"
        linkComponent={[Function]}
        target="_self"
      >
        <SocialIcon
          iconText=""
          iconType="snapchat"
        />
      </Styled(Link)>
    </li>
    <li
      key="Instagram"
    >
      <Styled(Link)
        href="//www.instagram.com/eurosport"
        linkComponent={[Function]}
        target="_self"
      >
        <SocialIcon
          iconText=""
          iconType="instagram"
        />
      </Styled(Link)>
    </li>
  </Styled(ul)>
</Styled(div)>
`;
